custom/icds_reports/utils/aggregation_helpers/distributed/agg_gov_dashboard.py

```diff
@@ -0,0 +1,194 @@
+                    valid_all_registered_in_month ELSE 0 END ) AS open_reg_in_month_0_3_male,
+
+                sum(CASE WHEN sex='F' AND age_tranche::INTEGER BETWEEN 37 AND 72 AND
+                    date_trunc('month',opened_on)=%(start_date)s THEN
+                    valid_in_month ELSE 0 END ) AS valid_reg_in_month_3_6_female,
+                sum(CASE WHEN sex='M' AND age_tranche::INTEGER BETWEEN 37 AND 72 AND
+                    date_trunc('month',opened_on)=%(start_date)s THEN
+                    valid_in_month ELSE 0 END ) AS valid_reg_in_month_3_6_male,
+
+                sum(CASE WHEN sex='F' AND age_tranche::INTEGER BETWEEN 37 AND 72 AND
+                    date_trunc('month',opened_on)=%(start_date)s THEN
+                    valid_all_registered_in_month ELSE 0 END ) AS  open_reg_in_month_3_6_female,
+
+                sum(CASE WHEN sex='M' AND age_tranche::INTEGER BETWEEN 37 AND 72 AND
+                    date_trunc('month',opened_on)=%(start_date)s THEN
+                    valid_all_registered_in_month ELSE 0 END ) AS  open_reg_in_month_3_6_male
+            FROM child_health_monthly
+            WHERE month=%(start_date)s
+            GROUP BY supervisor_id, awc_id;
+
+        UPDATE "{tablename}" agg_gov
+        SET total_0_3_female_benefit_till_date = ut.valid_all_0_3_female,
+            total_0_3_male_benefit_till_date = ut.valid_all_0_3_male,
+            total_0_3_female_reg_till_date = ut.open_all_0_3_female,
+            total_0_3_male_reg_till_date = ut.open_all_0_3_male,
+            total_3_6_female_benefit_till_date = ut.valid_all_3_6_female,
+            total_3_6_male_benefit_till_date = ut.valid_all_3_6_male,
+            total_3_6_female_reg_till_date = ut.open_all_3_6_female,
+            total_3_6_male_reg_till_date = ut.open_all_3_6_male,
+            total_0_3_female_benefit_in_month = ut.valid_reg_in_month_0_3_female,
+            total_0_3_male_benefit_in_month = ut.valid_reg_in_month_0_3_male,
+            total_0_3_female_reg_in_month = ut.open_reg_in_month_0_3_female,
+            total_0_3_male_reg_in_month = ut.open_reg_in_month_0_3_male,
+            total_3_6_female_benefit_in_month = ut.valid_reg_in_month_3_6_female,
+            total_3_6_male_benfit_in_month = ut.valid_reg_in_month_3_6_male,
+            total_3_6_female_reg_in_month = ut.open_reg_in_month_3_6_female,
+            total_3_6_male_reg_in_month = ut.open_reg_in_month_3_6_male
+        FROM temp_gov_dashboard ut
+        WHERE agg_gov.awc_id = ut.awc_id;
+
+        DROP TABLE temp_gov_dashboard;
+
+        """.format(
+            tablename=self._tablename_func(),
+        ), {
+            'start_date': self.month_start
+        }
+
+    def indexes(self):
+        """
+        Returns query to create index with columns month, state_id and awc_id
+        """
+        index_cols = ['month', 'state_id', 'awc_id']
+
+        return 'CREATE INDEX ON "{}" ({})'.format(self._tablename_func(), ', '.join(index_cols))
```

Comment:
given that there is nothing dynamic here, it would be a little easier to read if the columns where just written there.

Reply:
Sure.